File: app/src/main/java/com/garymcgowan/moviepedia/widget/VariableColumnGridLayoutManager.java
package com.garymcgowan.moviepedia.widget;

import android.content.Context;
import android.support.annotation.DimenRes;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;


public class VariableColumnGridLayoutManager extends GridLayoutManager {

    private float minItemWidthPixels;

    public VariableColumnGridLayoutManager(Context context, @DimenRes int minItemWidth) {
        super(context, 1);
        this.minItemWidthPixels = context.getResources().getDimension(minItemWidth);
    }

    @Override
    public void onLayoutChildren(RecyclerView.Recycler recycler,
                                 RecyclerView.State state) {
        updateSpanCount();
        super.onLayoutChildren(recycler, state);
    }

    private void updateSpanCount() {
        float spanCount = getWidth() / minItemWidthPixels;
        if (spanCount < 1) {
            spanCount = 1;
        }
        this.setSpanCount((int) spanCount);
    }
}

File: app/src/main/java/com/garymcgowan/moviepedia/widget/VariableColumnGridLayoutManager.kt
package com.garymcgowan.moviepedia.widget

import android.content.Context
import androidx.annotation.DimenRes
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView

class VariableColumnGridLayoutManager(context: Context, @DimenRes minItemWidth: Int) : GridLayoutManager(context, 1) {

    private val minItemWidthPixels: Float = context.resources.getDimension(minItemWidth)

    override fun onLayoutChildren(recycler: RecyclerView.Recycler?,
                                  state: RecyclerView.State) {
        updateSpanCount()
        super.onLayoutChildren(recycler, state)
    }

    private fun updateSpanCount() {
        var spanCount = width / minItemWidthPixels
        if (spanCount < 1) {
            spanCount = 1f
        }
        this.spanCount = spanCount.toInt()
    }
}
